classes/order/Order.php

```diff
@@ -926,8 +926,12 @@ public function isInPreparation()
      */
     public function isPaidAndShipped()
     {
-        $order_state = $this->getCurrentOrderState();
-        if ($order_state && $order_state->paid && $order_state->shipped) {
+        $result = Db::getInstance(_PS_USE_SQL_SLAVE_)->getRow('
+			SELECT SUM(os.`paid`) AS `paid`, SUM(os.`shipped`) AS `shipped` FROM `' . _DB_PREFIX_ . 'order_history` oh 
+			LEFT JOIN `' . _DB_PREFIX_ . 'order_state` os ON (os.`id_order_state` = oh.`id_order_state`) 
+			WHERE oh.`id_order`= ' . (int) $this->id );
+		
+        if ($result['paid'] > 0 && $result['shipped'] > 0) {
```

Comment:
🤔  basing yourself on the SUM may lead to errors IMHO

if I mark the order as paid and then mark it as unpaid, this method will return true

Reply:
Hi @sowbiba ,

as far as I know, it is **not** possible to mark the order as paid and then mark it as unpaid.
Once a customer has paid, I don't find why anyone should go and mark it as unpaid.

And anyway with this change the code will do exactly what it says in the RMA documentation:

https://doc.prestashop.com/display/PS17/Merchandise+Returns

> The orders need to have had at least the following statuses:
> At least two statuses: one with the condition "Set the order as paid" enabled and the second one with the option "Set the order as shipped" enabled.
> Or
> 
> One status which has both conditions ("Set the order as paid" and "Set the order as shipped") enabled.